### PR TITLE
:book: added AzureStackHCI infrastructure provider to list of providers in the CAPI book

### PR DESCRIPTION
**Pull request by daschott**
**What this PR does / why we need it**:
This docs PR adds a reference to the [AzureStackHCI infrastructure provider](https://github.com/microsoft/cluster-api-provider-azurestackhci) to the list of providers in the CAPI book.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/reference/providers.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ updated info about which API version they are supporting.
 - [Tencent Cloud](https://github.com/TencentCloud/cluster-api-provider-tencent)
 - [vSphere](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere)
 - [Alibaba Cloud](https://github.com/oam-oss/cluster-api-provider-alicloud)
+- [Azure Stack HCI](https://github.com/microsoft/cluster-api-provider-azurestackhci)
```
Comment (CecileRobertMichon) [inline docs/book/src/reference/providers.md:29]: not sure why Alibaba Cloud is at the bottom but the rest is alphabetically sorted so let's move this up to after Azure

Comment (daschott) [reply docs/book/src/reference/providers.md:29]: I assume it is OK if I move Alibaba cloud as well then so that it becomes consistent?

**Review by CecileRobertMichon — APPROVED** [review]
/lgtm
/assign @vincepri @fabriziopandini

**Pull request merged**